pkg/sql/plan/base_binder.go

```diff
@@ -923,6 +927,40 @@ func bindFuncExprImplByPlanExpr(name string, args []*Expr) (*plan.Expr, error) {
 			args[0].Typ = makePlan2Type(&returnType)
 		}
 	}
+
+	// rewrite some cast rule:  expr:  int32Col > 10,
+	// old rule: cast(int32Col as int64) >10 ,   new rule: int32Col > (cast 10 as int32)
+	switch name {
+	case "=", "<", "<=", ">", ">=", "<>":
+		// if constant's type higher than column's type
+		// and constant's value in range of column's type, then no cast was needed
```

Comment:
The right optimization rule:
if constant type's value is in range, then no cast was needed.
if constant type's value is not in range, then all these comparisons should be folded to a const true or false (or changed to is/isnot null).

Reply:
@aunjgr give the same suggestion. 
but that will do the rewrite for the function one by one. 
 "=" 's rule is different from ">"....  Maybe we can optimize it next time.